src/module/config.service.ts

```diff
@@ -155,15 +157,41 @@ export class ConfigService {
    * @returns {string}
    */
   static root(dir: string = ''): string {
-    return `${process.cwd()}${dir ? `/${dir}` : ''}`;
+    return path.resolve(process.cwd(), dir);
   }
 
   /**
    * @param {string} dir
    * @returns {string}
    */
   static src(dir: string = ''): string {
-    return ConfigService.root(`src${dir ? `/${dir}` : ''}`);
+    const srcPath = this.srcPath || this.root();
+    return path.resolve(srcPath, dir);
+  }
+
+  /**
+   * Resolves and stores sources directory for application.
+   * @param {string} startPath
+   *  The path for search starting. Can be any path under app sources path.
+   */
+  static resolveSrcPath(startPath: string): typeof ConfigService {
+    ok(path.isAbsolute(startPath), 'Start path must be an absolute path.');
```

Comment:
I wanna suggest to use `assert.ok(..)` instead.

It's more readable and idiomatic.

Reply:
updated